hazelcast/src/main/java/com/hazelcast/instance/AdvancedNetworkAddressPicker.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.instance;
+
+import com.hazelcast.config.AdvancedNetworkConfig;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.EndpointConfig;
+import com.hazelcast.config.InterfacesConfig;
+import com.hazelcast.config.ServerSocketEndpointConfig;
+import com.hazelcast.config.TcpIpConfig;
+import com.hazelcast.internal.cluster.impl.EndpointQualifier;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.nio.Address;
+
+import java.nio.channels.ServerSocketChannel;
+import java.util.HashMap;
+import java.util.Map;
+
+class AdvancedNetworkAddressPicker
```

Comment:
Some short javadoc describing the purpose and behaviour would be really helpful here.